js/libs/form.js

```diff
@@ -472,185 +474,239 @@ Fliplet.Widget.instance('form-builder', function(data) {
+  
+            if (data.generateEmailTemplate && data.onSubmit && data.onSubmit.indexOf('generateEmail') > -1) {
+              operation = Fliplet.Communicate.composeEmail(_.extend({}, data.generateEmailTemplate), formData);
+            }
+  
+            if (data.linkAction && data.redirect) {
+              return operation.then(function () {
+                return trackEventOp.then(function () {
+                  Fliplet.Navigate.to(data.linkAction);
+                });
+              }).catch(function (err) {
+                Fliplet.Modal.alert({
+                  message: Fliplet.parseError(err)
+                });
                 Fliplet.Navigate.to(data.linkAction);
-              });
-            }).catch(function (err) {
-              Fliplet.Modal.alert({
-                message: Fliplet.parseError(err)
-              });
-              Fliplet.Navigate.to(data.linkAction);
-            })
-          }
-
-          $vm.isSent = true;
-          $vm.isSending = false;
-          $vm.reset(false);
-          /**
-           * When we try to submit a form in Edge or IE11 and use components date picker and rich text
-           * (only in this sequence) we could saw that rich text textarea become empty but there was no
-           * message that we successfully submitted the form. That was because Vue wasn't updating view.
-           * $forceUpdate solve this issue.
-           */
-          $vm.$forceUpdate();
-
-          $vm.loadEntryForUpdate();
-        }, function(err) {
-          console.error(err);
-          $vm.error = Fliplet.parseError(err);
-          $vm.isSending = false;
-          Fliplet.Hooks.run('onFormSubmitError', { formData: formData, error: err });
+              })
+            }
+  
+            $vm.isSent = true;
+            $vm.isSending = false;
+            $vm.reset(false);
+            /**
+             * When we try to submit a form in Edge or IE11 and use components date picker and rich text
+             * (only in this sequence) we could saw that rich text textarea become empty but there was no
+             * message that we successfully submitted the form. That was because Vue wasn't updating view.
+             * $forceUpdate solve this issue.
+             */
+            $vm.$forceUpdate();
+  
+            $vm.loadEntryForUpdate();
+          }, function(err) {
+            console.error(err);
+            $vm.error = Fliplet.parseError(err);
+            $vm.isSending = false;
+            Fliplet.Hooks.run('onFormSubmitError', { formData: formData, error: err });
+          });
+  
+          // We might use this code to save the form data locally when going away from the page
+          // $(window).unload(function onWindowUnload() {
+          //   localStorage.setItem('fl-form-data-' + data.id, this.fields.map(function (field) {
+          //     return { name: field.name, value: field.value };
+          //   }));
+          // });
+        })
+        .catch(function() {
```

Comment:
Currently, if a form submission fails, e.g. no security permission, does the app user see a message? If so, will this silent `.catch()` end up silencing failed submissions? We don't want that.

Reply:
Reported [here](https://weboo.atlassian.net/browse/ID-1009)